fedn/fedn/network/api/client.py

```diff
@@ -0,0 +1,263 @@
+    :type host: str
+    :param port: The port of the api server.
+    :type port: int
+    :param secure: Whether to use https.
+    :type secure: bool
+    :param verify: Whether to verify the server certificate.
+    :type verify: bool
+    """
+
+    def __init__(self, host, port, secure=False, verify=False):
+        self.host = host
+        self.port = port
+        self.secure = secure
+        self.verify = verify
+
+    def _get_url(self, endpoint):
+        if self.secure:
+            protocol = 'https'
+        else:
+            protocol = 'http'
+        return f'{protocol}://{self.host}:{self.port}/{endpoint}'
+
+    def get_model_trail(self):
+        """ Get the model trail.
+
+        :return: The model trail as dict including commit timestamp.
+        :rtype: dict
+        """
+        response = requests.get(self._get_url('get_model_trail'), verify=self.verify)
+        return response.json()
+
+    def list_clients(self):
+        """ Get all clients from the statestore.
+
+        return: All clients.
+        rtype: dict
+        """
+        response = requests.get(self._get_url('list_clients'))
+        return response.json()
+
+    def get_active_clients(self):
+        """ Get all active clients from the statestore.
+
+        :param combiner_id: The combiner id to get active clients for.
+        :type combiner_id: str
+        :return: All active clients.
+        :rtype: dict
+        """
+        response = requests.get(self._get_url('get_active_clients'), verify=self.verify)
+        return response.json()
+
+    def list_combiners(self):
+        """ Get all combiners in the network.
+
+        :return: All combiners with info.
+        :rtype: dict
+        """
+        response = requests.get(self._get_url('list_combiners'))
+        return response.json()
+
+    def get_combiner(self, combiner_id):
+        """ Get a combiner from the statestore.
+
+        :param combiner_id: The combiner id to get.
+        :type combiner_id: str
+        :return: The combiner info.
+        :rtype: dict
+        """
+        response = requests.get(self._get_url(f'get_combiner?combiner={combiner_id}'), verify=self.verify)
+        return response.json()
+
+    def list_rounds(self):
+        """ Get all rounds from the statestore.
+
+        :return: All rounds with config and metrics.
+        :rtype: dict
+        """
+        response = requests.get(self._get_url('list_rounds'))
+        return response.json()
+
+    def get_round(self, round_id):
+        """ Get a round from the statestore.
+
+        :param round_id: The round id to get.
+        :type round_id: str
+        :return: The round config and metrics.
+        :rtype: dict
+        """
+        response = requests.get(self._get_url(f'get_round?round_id={round_id}'), verify=self.verify)
+        return response.json()
+
+    def start_session(self, session_id=None, round_timeout=180, rounds=5, round_buffer_size=-1, delete_models=True,
```

Comment:
It seems to not take in the parameters correctly, not that I here ser rounds and helper, but they were ignored: 

>>> client.start_session(round_timeout=30,rounds=1,helper='pytorchhelper')
{'config': {'clients_requested': 1, 'clients_required': 1, 'committed_at': '2023-08-16 05:58:29', 'delete_models_storage': True, 'helper_type': 'kerashelper', 'model_id': 'a1043100-e918-4855-a240-09a360ea76ca', 'round_timeout': 180.0, 'rounds': 5, 'session_id': '1fa4cd6d-53cd-41a6-b66f-ea6e609bd755', 'task': '', 'validate': True}, 'message': 'Session started successfully.', 'success': True}